pecos/xmc/base.py

```diff
@@ -1030,7 +1030,14 @@ def __len__(self):
         def __add__(self, other):
             if not isinstance(other, HierarchicalMLModel.PredParams):
                 other = HierarchicalMLModel.PredParams(other)
-            ret_model_chain = self.model_chain + other.model_chain
+
+            if isinstance(self.model_chain, type(other.model_chain)):
```

Comment:
can be simplified to ret_model_chain = tuple(self.model_chain) + tuple(other.model_chain)